graphql_compiler/tests/snapshot_tests/test_cost_estimation.py

```diff
@@ -1398,3 +1406,101 @@ def test_disjoint_intervals(self) -> None:
         self.assertEqual(expected_intersection, received_intersection)
+
+    @pytest.mark.usefixtures("snapshot_orientdb_client")
+    def test_int_value_conversion_uuid(self):
+        schema_graph = generate_schema_graph(self.orientdb_client)  # type: ignore  # from fixture
+        graphql_schema, type_equivalence_hints = get_graphql_schema_from_schema_graph(schema_graph)
+        pagination_keys = {vertex_name: "uuid" for vertex_name in schema_graph.vertex_class_names}
+        uuid4_fields = {vertex_name: {"uuid"} for vertex_name in schema_graph.vertex_class_names}
+        statistics = LocalStatistics({})
+        schema_info = QueryPlanningSchemaInfo(
+            schema=graphql_schema,
+            type_equivalence_hints=type_equivalence_hints,
+            schema_graph=schema_graph,
+            statistics=statistics,
+            pagination_keys=pagination_keys,
+            uuid4_fields=uuid4_fields,
+        )
+
+        uuid_values = [
+            "00000000-0000-0000-0000-000000000000",
+            "80000000-0000-0000-0000-000000000000",
+            "80000000-0000-0000-0000-000000000001",
+            "ffffffff-ffff-ffff-ffff-ffffffffffff",
+        ]
+        for uuid_value in uuid_values:
+            int_value = convert_field_value_to_int(schema_info, "Event", "uuid", uuid_value)
+            recovered_uuid = convert_int_to_field_value(schema_info, "Event", "uuid", int_value)
+            self.assertEqual(uuid_value, recovered_uuid)
+
+        invalid_uuid_values = [
+            "80000000-0000-",
+        ]
+        for uuid_value in invalid_uuid_values:
+            with self.assertRaises(Exception):
+                int_value = convert_field_value_to_int(schema_info, "Event", "uuid", uuid_value)
+
+    @pytest.mark.usefixtures("snapshot_orientdb_client")
+    def test_int_value_conversion_datetime(self):
+        schema_graph = generate_schema_graph(self.orientdb_client)  # type: ignore  # from fixture
+        graphql_schema, type_equivalence_hints = get_graphql_schema_from_schema_graph(schema_graph)
+        pagination_keys = {vertex_name: "uuid" for vertex_name in schema_graph.vertex_class_names}
+        uuid4_fields = {vertex_name: {"uuid"} for vertex_name in schema_graph.vertex_class_names}
+        statistics = LocalStatistics({})
+        schema_info = QueryPlanningSchemaInfo(
+            schema=graphql_schema,
+            type_equivalence_hints=type_equivalence_hints,
+            schema_graph=schema_graph,
+            statistics=statistics,
+            pagination_keys=pagination_keys,
+            uuid4_fields=uuid4_fields,
+        )
+
+        datetime_values = [
+            datetime(2000, 1, 1),
+            datetime(3000, 1, 1, tzinfo=None),
+            datetime(1000, 1, 1, tzinfo=pytz.utc),
+            datetime(1, 1, 1, tzinfo=pytz.utc),
+            datetime(2000, 1, 1, 20, 55, 40, 877633, tzinfo=pytz.utc),
+            datetime(2000, 1, 1, 20, 55, 40, 877633, tzinfo=pytz.timezone("GMT")),
+        ]
```

Comment:
How about also adding a non-UTC and non-UTC-equivalent timezone as a test case?
```suggestion
            datetime(2000, 1, 1, 20, 55, 40, 877633, tzinfo=pytz.timezone("America/New_York")),
        ]
```